nexus/src/app/instance.rs

```diff
@@ -1017,4 +1017,39 @@ impl super::Nexus {
             }
         }
     }
+
+    /// Returns the requested range of serial console output bytes,
+    /// provided they are still in the sled-agent's cache.
+    pub(crate) async fn instance_serial_console_data(
+        &self,
+        opctx: &OpContext,
+        organization_name: &Name,
+        project_name: &Name,
+        instance_name: &Name,
+        params: &nexus::InstanceSerialConsoleRequest,
+    ) -> Result<nexus::InstanceSerialConsoleData, Error> {
+        let db_instance = self
+            .instance_fetch(
+                opctx,
+                organization_name,
+                project_name,
+                instance_name,
+            )
+            .await?;
+
+        let sa = self.instance_sled(&db_instance).await?;
+        let data = sa
+            .instance_serial_get(
+                &db_instance.identity().id,
+                params.byte_offset.map(|x| x as i32),
```

Comment:
I'm assuming this is because we use `usize`/`isize` in the rust types but the generated OpenAPI stuff maps that to `u32`/`i32`? If so, would it be clearer to just use the explicitly sized ones as well in the types?

Reply:
yeah, this was the rustaceanary urge to use usize/isize for array length / indexing, but if that's the right thing to do in the OpenAPI world i can certainly change that and do the conversions in the other direction where needed.